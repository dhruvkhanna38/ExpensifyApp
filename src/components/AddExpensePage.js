import React from "react";
import {ExpenseForm} from "./ExpenseForm"
import {connect} from "react-redux";
import {addExpense} from "../actions/expenses";

export const AddExpensePage = (props)=>{
   return (
      <div>
         Create Component
         <ExpenseForm onSubmit={(expense)=>{
            props.dispatch(addExpense(expense))
            props.history.push("/");
         }}></ExpenseForm>
      </div>
   )
}

export const ConnectedAddExpensePage = connect()(AddExpensePage)