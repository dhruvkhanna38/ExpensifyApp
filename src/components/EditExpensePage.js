import React from "react";
import {connect} from "react-redux";
import {ExpenseForm} from "./ExpenseForm";
import {editExpense, removeExpense} from '../actions/expenses'

export const EditExpensePage = (props)=>{
   return (
      <div>
         {console.log(props.expense)}
         <ExpenseForm 
         expense= {props.expense}
         onSubmit= {
            (expense)=>{
               props.dispatch(editExpense(props.expense.id, expense))
               props.history.push('/')
            }
         }>
         </ExpenseForm>
         <button onClick={(e)=>{
            props.dispatch(removeExpense({id:props.expense.id}))
            props.history.push('/')
         }}>Remove</button>
      </div>
   )
}

const mapStateToProps = (state, props)=>{
   return {
      expense: state.expenses.find((expense)=>{
         return expense.id === props.match.params.id
      })
   }
}

export const ConnectedEditExpensePage = connect(mapStateToProps)(EditExpensePage);