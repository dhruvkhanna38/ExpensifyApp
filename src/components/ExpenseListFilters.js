import React from "react";
import {connect} from "react-redux";
import {setTextFilter, sortByAmount, sortByDate, setStartDate, setEndDate} from "../actions/filters";
import {DateRangePicker, DateRangePickerInput} from "react-dates";

class ExpenseListFilters extends React.Component{

   state = {
      calenderFocused: null,

   }

   onDatesChange = ({startDate, endDate})=>{
      this.props.dispatch(setStartDate(startDate));
      this.props.dispatch(setEndDate(endDate));
   }

   onFocusChange = (calenderFocused)=>{
      this.setState(()=>{
         return {
            calenderFocused
         }
      })

   }

   render(){
      return (
         <div>
            <input type="text" value={this.props.filters.text} onChange={(e)=>{
               this.props.dispatch(setTextFilter(e.target.value));
            }}></input>
   
            <select onChange={(e)=>{
               if(e.target.value === 'date'){
                  this.props.dispatch(sortByDate())
               }else if (e.target.value === 'amount'){
                  this.props.dispatch(sortByAmount())
               }
               console.log(e.target.value);
            }}>
               <option value="date">Date</option>
               <option value="amount">Amount</option>
            </select>
            <DateRangePicker startDate={this.props.filters.startDate} endDate={this.props.filters.endDate}
                             onDatesChange={this.onDatesChange} focusedInput={this.state.calenderFocused}
                             onFocusChange={this.onFocusChange} numberOfMonths={1} isOutsideRange={()=>false} showClearDates={true}>
            </DateRangePicker>
         </div>
      )
   }
}


export const ConnectedExpenseListFilters = connect((state)=>{
   return {
      filters: state.filters
   }
})(ExpenseListFilters);