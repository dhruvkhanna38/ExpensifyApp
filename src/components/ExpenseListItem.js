import React from "react";
import {removeExpense} from "../actions/expenses";
import {connect} from "react-redux";
import {Link} from 'react-router-dom';

export const ExpenseListItem = (props)=>{
   return (
      <div>
         <Link to={`/edit/${props.id}`}><p>{props.description}</p></Link>
         <h3>{props.amount} - {props.createdAt}</h3>
      </div>
   )
}

export const ConnectedExpenseListItem = connect((state)=>{
   return {
      expenses: state.expenses
   }
})(ExpenseListItem)