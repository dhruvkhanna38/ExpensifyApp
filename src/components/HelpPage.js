
import React from "react";

export const HelpPage = ()=>{
   return (
      <div>
         Help Component
      </div>
   )
}