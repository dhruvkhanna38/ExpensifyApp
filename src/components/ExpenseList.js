import React from "react";
import {connect} from "react-redux";
import {ConnectedExpenseListItem} from "./ExpenseListItem"
import {getVisibleExpenses} from "../selectors/expenses";
import {ConnectedExpenseListFilters} from "./ExpenseListFilters";

export const ExpenseList = (props)=>{
   return (
      <div>
         { props.expenses.length===0 ? 
           (<p>No Expenses</p>):(
           props.expenses.map((expense)=>{
               return <ConnectedExpenseListItem key={expense.id} {...expense}></ConnectedExpenseListItem>
           }))
         }
         <ConnectedExpenseListFilters></ConnectedExpenseListFilters>
      </div>
   )
}

export const ConnectedExpenseList = connect((state)=>{
   return {
      expenses: getVisibleExpenses(state.expenses, state.filters)
   }
})(ExpenseList)

