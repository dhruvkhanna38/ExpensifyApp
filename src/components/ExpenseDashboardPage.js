import React from "react";
import{ConnectedExpenseList} from "./ExpenseList";

export const ExpenseDashboardPage =()=> (
   <div>
      Dashboard Component
      <ConnectedExpenseList></ConnectedExpenseList>
   </div>
)

