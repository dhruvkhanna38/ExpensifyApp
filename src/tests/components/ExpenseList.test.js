import React from "react";
import {shallow} from 'enzyme';
import {ExpenseList} from "../../components/ExpenseList";
import {expenses} from "../fixtures/expenses";


test("should render ExpenseList with expenses", ()=>{
   const wrapper = shallow(<ExpenseList expenses={expenses}></ExpenseList>)
   expect(wrapper).toMatchSnapshot()
})

test("shpould render expense List with empty array", ()=>{
   const wrapper = shallow(<ExpenseList expenses={[]}></ExpenseList>)
   expect(wrapper).toMatchSnapshot()
})