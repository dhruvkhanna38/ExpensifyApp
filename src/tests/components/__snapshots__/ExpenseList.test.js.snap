// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseList with expenses 1`] = `
<div>
  <Connect(ExpenseListItem)
    amount={195}
    createdAt={0}
    description="Gum"
    id="1"
    key="1"
    note=""
  />
  <Connect(ExpenseListItem)
    amount={29500}
    createdAt={-345600000}
    description="Rent"
    id="2"
    key="2"
    note=""
  />
  <Connect(ExpenseListItem)
    amount={200000}
    createdAt={345600000}
    description="Credit Card"
    id="3"
    key="3"
    note=""
  />
  <Connect(ExpenseListFilters) />
</div>
`;

exports[`shpould render expense List with empty array 1`] = `
<div>
  <p>
    No Expenses
  </p>
  <Connect(ExpenseListFilters) />
</div>
`;
